web/src/users/signup.rs

```diff
@@ -0,0 +1,70 @@
+use axum::{
+    extract::State,
+    response::{IntoResponse, Redirect, Response},
+    Form,
+};
+use cja::app_state::AppState as _;
+use db::users::User;
+
+use password_auth::generate_hash;
+
+use crate::{sessions::Session, templates::Template, AppState, WebResult};
+
+pub async fn get(t: Template) -> impl IntoResponse {
+    t.render(maud::html! {
+      form method="post" action="/signup" {
+        input type="text" name="username" placeholder="Username";
+        input type="password" name="password" placeholder="Password";
+        input type="submit" value="Signup";
+      }
+    })
+}
+
+#[derive(serde::Deserialize)]
+pub struct FormData {
+    username: String,
+    password: String,
+}
+
+pub async fn post(
+    session: Session,
+    State(state): State<AppState>,
+    form_data: Form<FormData>,
+) -> WebResult<Response> {
```

Comment:
And here we create a new user instead of logging one in!

We DO also log you in after you signup